tools/write_pyi.py

```diff
@@ -168,17 +188,26 @@ def _formatannotation(annotation, base_module=None):
         retval = re.sub("NoneType", "None", retval)
         return retval
 
+    def _formatvalue(value):
+        return "=" + ("..." if value is Ellipsis else repr(value))
+
     argspec = inspect_formatargspec(
         *spec,
         formatannotation=_formatannotation,
+        formatvalue=_formatvalue,
         formatreturns=lambda val: f"-> {_formatannotation(val)}",
     )
+
+    overload = "@overload" if is_overload else ""
     contextmanager = "@contextmanager" if is_context_manager else ""
     func_text = textwrap.dedent(
         f"""
+    {overload}
     {contextmanager}
-    def {name}{argspec}:
-        '''{fn.__doc__}'''
+    def {name}{argspec}: {"..." if (is_overload or fn.__doc__ is None) else ""}
+        {"" if (is_overload or fn.__doc__ is None)
+        else "'''" + fn.__doc__ + os.linesep + "'''"
```

Comment:
it probably makes sense to create variables for this if an the above one

Reply:
reworked